### ISSUE TITLE
Has Invite Module flaw been fixed in V4?

### ISSUE THREAD

**Issue by phpfoxmetro** [moxi9/phpfox#259]
Disclaimer: I have not installed the v4 beta so I don't know if this has been fixed or not... But here you go anyway... 

Since the invention of the Invite Module, there was a fundamental flaw in the way phpFox handled Invites sent from your site to non-members. Those of us who have worked in the email and messaging space tried to bring this up but got the infamous - "working as designed". 

Here is the issue... 
PhpFox tries to send out invitations using the senders email address, as opposed to the systems default email system email address. Majority of email servers do: 

SPF (Sender Policy Framework) - https://en.wikipedia.org/wiki/Sender_Policy_Framew... 
DomainKeys - https://en.wikipedia.org/wiki/DomainKeys_Identifie... 
rDNS lookups - https://en.wikipedia.org/wiki/Reverse_DNS_lookup 

In essence, lets assume that your pretty phpFox site is http://www.contoso.com and you promote the use of the invite module, and users send tens of billions of emails from your site, chances are 90% of them will: 

a) either be quietly rejected, 
b) noisily be rejected and your domain going into a blacklist 
c) soft fail and end up in the junk folder 

The reason is simple, when an invite is sent out from your pretty dead site, it goes out as, e.g yourcreativeemailaddress@yahooooooo.com. When the receiving server does a SPF, DomainKey, or rDNS it sees that contoso.com is not authorized to send messages for @yahooooooo.com domain. 

In other words, this email message is a spoof and should be crushed like a roach, and your phpFox site gobbled by the dooms of hell for being an innocent spammer!

**Comment by Data66PF** [comment]
marked a bug

**Comment by RaymondBenc** [comment]
Fix is in v4 and is actually in v3 as well:
https://github.com/moxi9/phpfox-v3-dist/blob/master/upload/module/invite/include/component/controller/index.class.php#L61

Make sure to update your "Email" setting with the email you want to send it from, otherwise it falls back to the user.

**Comment by Data66PF** [comment]
Made a tutorial http://scripttechs.com/article/360/v3-mail-setting-phrase-wrong-do-this-to-get-your-emails-delivered/ as the phrasing in it makes one think that should be the site name. It really should be changed.

From:
This is the name displayed when users receive emails from this site. 

Would be better as:

From:
This is the email displayed when users receive emails from this site.
